write-rules.js

```diff
@@ -0,0 +1,172 @@
+const fs = require('node:fs');
+const stylelint = require('stylelint');
+
+const COMMENT = '<!-- RULES:START -->';
+const README = 'README.md';
+
+const getUrl = (rule) => {
+  const parts = rule.split('/');
+
+  if (parts.length === 1) {
+    return `https://stylelint.io/user-guide/rules/list/${rule}/`;
+  }
+
+  const [group, ruleName] = parts;
+
+  const nonStylelintRules = {
+    'order': () =>
+      `https://github.com/hudochenkov/stylelint-order/blob/master/rules/${ruleName}/README.md`,
+    'scale-unlimited': () =>
+      'https://github.com/AndyOGo/stylelint-declaration-strict-value',
+    'scss': () =>
+      `https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/${ruleName}/README.md`,
+  };
+
+  return (nonStylelintRules[group] || (() => null))();
+};
+
+/**
+ * Attempts to parse the config value to return a pretty formatted version.
+ * If the value becomes too long it will instead direct the viewer to an external link.
+ *
+ * @param {array} config
+ * @param {object?} options
+ * @returns {string}
+ */
+const getPrettyConfig = (
+  config,
+  { defaultValue = 'Enabled', maxLength = 75 } = {},
+) => {
+  if (!Array.isArray(config)) return defaultValue;
+
+  const configItems = config.filter((item) => typeof item !== 'boolean');
+
+  const prettyConfig = configItems.map(JSON.stringify).join(', ');
+
+  if (prettyConfig.length === 0) return defaultValue;
+
+  if (prettyConfig.length > maxLength) {
+    return `${defaultValue} - [see Config][config]`;
+  }
+
+  return `\`${prettyConfig}\``;
+};
+
+/**
+ * Prepares a reasonably formatted markdown table based on the spec
+ * https://github.github.com/gfm/#tables-extension-
+ *
+ * Ensuring that column widths remain readable while viewing the markdown raw.
+ *
+ * If a rule URL can be resolved, the rule will be added with a link to it, otherwise
+ * the rule will be just shown as is.
+ *
+ * Rule configuration is not yet included in the table.
+ *
+ * @param {Object} rules
+ * @returns {String}
+ */
+const createMarkdownTable = (rules) => {
+  const enabledRules = Object.entries(rules)
+    .filter(([, value]) => value !== null)
+    .sort(([ruleA], [ruleB]) => ruleA.localeCompare(ruleB));
```

Comment:
I find it a bit unexpected that rules from plugins are in-between core rules. Though not sure how else I’d prefer it to be.

Reply:
yeah, I just went alpha because it was simpler initially.